src/main/java/net/snowflake/ingest/streaming/InsertValidationResponse.java

```diff
@@ -125,6 +137,18 @@ public void setMissingNotNullColNames(List<String> missingNotNullColNames) {
       this.missingNotNullColNames = missingNotNullColNames;
     }
 
+    /**
+     * Add a missing non-nullable column name in the input row compared with the table schema
+     *
+     * @param missingNotNullColName the missing non-nullable column name
+     */
+    public void addMissingNotNullColName(String missingNotNullColName) {
```

Comment:
any chance this can be called from multiple threads?

Reply:
Insert error instance are independent between rows, I think this should be safe.